src/shared/port-provider.js

```diff
@@ -71,23 +71,23 @@ export class PortProvider {
     this._emitter = new TinyEmitter();
 
     /**
-     * Map of source frame to channels that have been created for this frame.
+     * IDs of port requests that have been handled.
      *
      * This is used to avoid responding to the same request multiple times.
      * Guest frames in particular may send duplicate requests (see comments in
      * PortFinder).
      *
-     * @type {WeakMap<Window, Set<Channel>>}
+     * @type {Set<string>}
      */
-    this._sentChannels = new WeakMap();
+    this._handledRequests = new Set();
 
     // Create the `sidebar-host` channel immediately, while other channels are
     // created on demand
     this._sidebarHostChannel = new MessageChannel();
 
     this._listeners = new ListenerCollection();
 
-    /** @type {Array<Message & {allowedOrigin: string}>} */
+    /** @type {Array<Partial<Message> & {allowedOrigin: string}>} */
```

Comment:
`Partial` is needed because `_allowedMessages` doesn't include the `requestId` field, since that changes with each request.